### PR TITLE
create image upload

### PR DESCRIPTION
**Pull request by niklasrammelkamp**
Hi, here's my pull request to the following User Story:
[image upload](https://github.com/niklasrammelkamp/capstone-project/issues/11#issue-1557928734)

### CODE REVIEW

File: components/UploadForm/index.js [hunk 1]
```diff
@@ -41,19 +37,33 @@ export default function UploadForm({ initialCategories, onSubmit }) {
...
     const currentDate = new Date().toLocaleDateString("us-US", {
       dateStyle: "medium",
     });
 
+    // creating the new Picture Object
     const uploadObject = {
       id: crypto.randomUUID(),
-      image: data.image,
+      image: imageData.secure_url,
       describtion: data.describtion,
```
Comment (marcterre) [inline components/UploadForm/index.js:67]: typo in describtion, write it like that -> description

File: components/UploadForm/index.js [hunk 2]
```diff
@@ -77,7 +87,7 @@ export default function UploadForm({ initialCategories, onSubmit }) {
   return (
     <form onSubmit={handleSubmit}>
       <label htmlFor="name">upload image</label>
-      <input type="text" id="image" name="image" required />
+      <input type="file" id="image" name="imageFile" required />
       <br />
       <label htmlFor="describtion">describtion</label>
```
Comment (marcterre) [inline components/UploadForm/index.js:92]: Why break the line? You can do that with styling your form :)
also a typo in description

**Review by marcterre — APPROVED** [review]
Looks good, looking forward to your preview! Saw just a small typo
have fun ✌️

File: components/UploadForm/index.js (hunk 2 above)
Comment (niklasrammelkamp) [reply components/UploadForm/index.js:92]: Oh f***, I think I wrote 'description' everywhere wrong. I'll try to fix it.

Comment (niklasrammelkamp) [reply components/UploadForm/index.js:92]: The breaks I wanna change, when I start styling.

File: components/UploadForm/index.js [hunk 3]
```diff
@@ -41,20 +37,34 @@ export default function UploadForm({ initialCategories, onSubmit }) {
...
-  function handleSubmit(event) {
+  // ----------- HANDLESUBMIT ----------------------------------
+  async function handleSubmit(event) {
     event.preventDefault();
 
     const formData = new FormData(event.target);
     const data = Object.fromEntries(formData);
 
+    setStatusIcon("... uploading image ⏳");
+    // uploading th image file
```
Comment (SaschaRadojkovic) [inline components/UploadForm/index.js:48]: missin an e in the

File: components/UploadForm/index.js [hunk 4]
```diff
@@ -41,20 +37,34 @@ export default function UploadForm({ initialCategories, onSubmit }) {
...
+    const response = await fetch("/api/upload", {
+      method: "POST",
+      body: formData,
+    });
+
+    // getting the uplaoded file
+    const imageData = await response.json();
+    console.log(imageData);
```
Comment (SaschaRadojkovic) [inline components/UploadForm/index.js:56]: i think you should remove the console.log

File: components/UploadForm/index.js [hunk 5]
```diff
@@ -77,10 +87,10 @@ export default function UploadForm({ initialCategories, onSubmit }) {
...
       <label htmlFor="name">upload image</label>
-      <input type="text" id="image" name="image" required />
+      <input type="file" id="image" name="imageFile" required />
       <br />
-      <label htmlFor="describtion">describtion</label>
-      <textarea type="text" id="describtion" name="describtion" required />
+      <label htmlFor="description">description</label>
+      <textarea type="text" id="description" name="description" required />
 
       {/* ------- serach function + suggestions ------- */}
```
Comment (SaschaRadojkovic) [inline components/UploadForm/index.js:95]: here a little typo serach instead of search

File: components/UploadForm/index.js [hunk 6]
```diff
@@ -90,27 +100,30 @@ export default function UploadForm({ initialCategories, onSubmit }) {
           type="text"
           value={inputCategoryValue}
           onChange={onChangeInput}
-          onBlur={() => setIsCategoryFound(false)}
           style={{
             outline: noCategoriesSelected && "red 2px solid",
           }}
```
Comment (SaschaRadojkovic) [inline components/UploadForm/index.js:105]: i think you could exchange this style with  styled-components `<StyledInput  isRed={noCategoriesSelected}>` and `const StyledInput =styled.input`
outline: ${(props) => (props.isRed &&  "red 2px solid" )}
``i hope this helps and is correct  i am not shure about the quotes after the && :)   but i think it is your only style till now and its only about mixing styles later

**Review by SaschaRadojkovic — APPROVED** [review]
good to go

**Pull request merged**